src/main/java/duke/command/Duke.java

```diff
@@ -0,0 +1,231 @@
+            "I am sorry, but I do not know what do you mean. " +
+                    "Please key in a valid input.";
+    static final String GREETINGS = LINE_SEPARATOR
+            + SPACING + "Hello! I'm Duke\n"
+            + SPACING + "What can I do for you?\n"
+            + LINE_SEPARATOR;
+    static final String NO_ARGUMENT_1 = "The ";
+    static final String NO_ARGUMENT_2 = " command is incomplete.";
+    static ArrayList<Task> taskList = new ArrayList<>();
+
+    public static void main(String[] args) {
+        readData();
+        System.out.println(GREETINGS);
+        String line;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            line = in.nextLine();
+            if (line.equals("bye")) {
+                break;
+            }
+            switch (line.split(" ")[0]) {
+            case "list":
+                list();
+                break;
+            case "done":
+                try {
+                    int currentTask = Integer.parseInt(line.split(" ")[1]) - 1;
+                    taskList.get(currentTask).markAsDone();
+                    String update = LINE_SEPARATOR
+                            + SPACING + "Nice! I've marked this task as done: \n"
+                            + SPACING + taskList.get(currentTask).toString() + "\n"
+                            + LINE_SEPARATOR;
+                    System.out.println(update);
+                    writeData();
+                } catch (IndexOutOfBoundsException e) {
+                    String error = LINE_SEPARATOR
+                            + SPACING + NO_ARGUMENT_1 + "done" + NO_ARGUMENT_2 + "\n"
+                            + LINE_SEPARATOR;
+                    System.out.println(error);
+                }
+                break;
+            case "deadline":
+                try {
+                    String taskName = line.substring("deadline ".length(), line.indexOf("/by "));
+                    String taskDdl = line.substring(line.indexOf("/by ") + "/by ".length());
+                    taskList.add(new Deadline(taskName, taskDdl));
+                    String response = LINE_SEPARATOR
+                            + SPACING + "added: " + taskList.get(taskList.size() - 1) + "\n"
+                            + LINE_SEPARATOR;
+                    System.out.println(response);
+                    writeData();
+                } catch (StringIndexOutOfBoundsException e) {
+                    String error = LINE_SEPARATOR
+                            + SPACING + NO_ARGUMENT_1 + "deadline" + NO_ARGUMENT_2 + "\n"
+                            + LINE_SEPARATOR;
+                    System.out.println(error);
+                }
+
+                break;
+            case "event":
+                try {
+                    String taskName = line.substring("event ".length(), line.indexOf("/at "));
+                    String taskTime = line.substring(line.indexOf("/at ") + "/at ".length());
```

Comment:
This naming is inconsistent with the previous "taskDdl". I recommending changing either one of them so that the names make more sense (perhaps taskDdl).